src/objects/GameObject.cpp

```diff
@@ -332,7 +297,12 @@ void GameObject::setRotation(chag::Quaternion r) {
     hasRotation = changed = true;
 }
 
-void GameObject::addChild(GameObject* child) {
+void GameObject::addChild(std::shared_ptr<GameObject> child) {
+    if (child->parent) {
+        throw std::invalid_argument("Given child already has a parent");
+    } else {
+        child->parent = shared_from_this();
+    }
```

Comment:
Perhaps this method should be renamed? It does more than adding children. Something like connectChildAndSetThisAsParent? Long, but at least it's very descriptive